kong/pdk/service.lua

```diff
@@ -109,146 +107,144 @@ local function new()
   end
 
 
-  if is_http_subsystem then
-    local tls = require("resty.kong.tls")
-
-    local set_upstream_cert_and_key = tls.set_upstream_cert_and_key
-    local set_upstream_ssl_verify = tls.set_upstream_ssl_verify
-    local set_upstream_ssl_verify_depth = tls.set_upstream_ssl_verify_depth
-    local set_upstream_ssl_trusted_store = tls.set_upstream_ssl_trusted_store
-
-    ---
-    -- Sets the client certificate used while handshaking with the Service.
-    --
-    -- The `chain` argument is the client certificate and intermediate chain (if any)
-    -- returned by functions such as [ngx.ssl.parse\_pem\_cert](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/ssl.md#parse_pem_cert).
-    --
-    -- The `key` argument is the private key corresponding to the client certificate
-    -- returned by functions such as [ngx.ssl.parse\_pem\_priv\_key](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/ssl.md#parse_pem_priv_key).
-    --
-    -- @function kong.service.set_tls_cert_key
-    -- @phases `rewrite`, `access`, `balancer`
-    -- @tparam cdata chain The client certificate chain
-    -- @tparam cdata key The client certificate private key
-    -- @treturn boolean|nil `true` if the operation succeeded, `nil` if an error occurred
-    -- @treturn string|nil An error message describing the error if there was one
-    -- @usage
-    -- local chain = assert(ssl.parse_pem_cert(cert_data))
-    -- local key = assert(ssl.parse_pem_priv_key(key_data))
-    --
-    -- local ok, err = kong.service.set_tls_cert_key(chain, key)
-    -- if not ok then
-    --   -- do something with error
-    -- end
-    service.set_tls_cert_key = function(chain, key)
-      check_phase(access_and_rewrite_and_balancer)
-
-      if type(chain) ~= "cdata" then
-        error("chain must be a parsed cdata object", 2)
-      end
-
-      if type(key) ~= "cdata" then
-        error("key must be a parsed cdata object", 2)
-      end
-
-      local res, err = set_upstream_cert_and_key(chain, key)
-      return res, err
+  local tls = require("resty.kong.tls")
```

Comment:
There's nothing better than the feeling of getting to de-indent a whole bunch of lines :D